python/ql/src/experimental/dataflow/internal/DataFlowPublic.qll

```diff
@@ -148,6 +149,19 @@ class ParameterNode extends EssaNode {
   override DataFlowCallable getEnclosingCallable() { this.isParameterOf(result, _) }
 }
 
+/**
+ * A node that controls whether other nodes are evaluated.
+ */
+class GuardNode extends ControlFlowNode {
+  ConditionBlock conditionBlock;
+
+  GuardNode() { this = conditionBlock.getLastNode() }
+
+  predicate controlsBlock(BasicBlock b, boolean testIsTrue) {
+    conditionBlock.controls(b, testIsTrue)
+  }
```

Comment:
Seems like I overlooked one :smile: 

```suggestion
  predicate controlsBlock(BasicBlock b, boolean branch) {
    conditionBlock.controls(b, branch)
  }
```

Reply:
we should also have a comment for this predicate, so I added one (not that it says much more than the predicate name)